common/djangoapps/third_party_auth/views.py

```diff
@@ -36,3 +41,15 @@ def saml_metadata_view(request):
     if not errors:
         return HttpResponse(content=metadata, content_type='text/xml')
     return HttpResponseServerError(content=', '.join(errors))
+
+
+@csrf_exempt
+@psa('{0}:complete'.format(URL_NAMESPACE))
```

Comment:
I'm unfamiliar with the `@psa` decorator. Probably worth a brief comment explaining what this is.